wtforms/fields/core.py

```diff
@@ -915,16 +915,23 @@ def validate(self, form, extra_validators=tuple()):
         of its own validators.
         """
         self.errors = []
+        error_occured = False
 
         # Run validators on all entries within
         for subfield in self.entries:
             if not subfield.validate(form):
                 self.errors.append(subfield.errors)
+                error_occured = True
+            else:
+                self.errors.append(None)
+
+        if all(x is None for x in self.errors):
+            self.errors = []
 
         chain = itertools.chain(self.validators, extra_validators)
         self._run_validation_chain(form, chain)
 
-        return len(self.errors) == 0
+        return not error_occured
```

Comment:
Hmmm. Yes, it will work, I guess.
Should I add the changes to this pull request? I'm still bad with pull requests, so I'm not sure it's even possible in technical meaning :)